liquidctl/driver/asus_gpu.py

```diff
@@ -0,0 +1,184 @@
+        FIXED = (0x01, 1)
+        BREATHING = (0x02, 1)
+        FLASH = (0x03, 1)
+        RAINBOW = (0x04, 0)
+
+        def __str__(self):
+            return self.name.capitalize()
+
+    
+
+    @classmethod
+    def probe(cls, smbus, vendor=None, product=None, address=None, match=None,
+              release=None, serial=None, **kwargs):
+
+        if (vendor and vendor != _ASUS) \
+                or (address and int(address, base=16) not in cls.ADDRESSES) \
+                or smbus.parent_subsystem_vendor != _ASUS \
+                or smbus.parent_vendor != _NVIDIA \
+                or smbus.parent_driver != 'nvidia' \
+                or release or serial:  # will never match: always None
+            return
+
+        supported = [
+            (_ROG_RTX_2080_TI, _RTX_2080_TI, "ASUS ROG RTX 2080ti (experimental)"),
+        ]
+
+        for (dev_id, sub_dev_id, desc) in supported:
+            if (product and product != sub_dev_id) \
+                    or (match and match.lower() not in desc.lower()) \
+                    or smbus.parent_subsystem_device != sub_dev_id \
+                    or smbus.parent_device != dev_id \
+                    or not smbus.description.startswith('NVIDIA i2c adapter 1 '):
+                continue
+
+            for address in cls.ADDRESSES:
+                val1=0
+                val2=0
+                try:
+                    smbus.open()
+                    val1 = smbus.read_byte_data(address, 0x20)
+                    val2 = smbus.read_byte_data(address, 0x21)
+                    smbus.close()
+                except:
+                    _LOGGER.debug(f'Device not found at {address}')
+
+                if val1 << 8 | val2 == cls._ASUS_GPU_MAGIC_VALUE:
+                    dev = cls(smbus, desc, vendor_id=_ASUS, product_id=_RTX_2080_TI,
+                        address=address)
+                    _LOGGER.debug(f'instanced driver for {desc} at address {address}')
+                    yield dev
+
+    def get_status(self, verbose=False, unsafe=None, **kwargs):
+        """Get a status report.
+
+        Returns a list of `(property, value, unit)` tuples.
+        """
+
+        # only RGB lighting information can be fetched for now; as that isn't
+        # super interesting, only enable it in verbose mode
+
+        if not verbose:
+            return []
+
+        if not (unsafe and 'rog_strix' in unsafe):
+            _LOGGER.warning('Device requires `rog_strix` unsafe flag')
+            return []
+
+        mode =  self._smbus.read_byte_data(self._address, self.REG_MODE)
+        red = self._smbus.read_byte_data(self._address, self.REG_RED)
+        blue = self._smbus.read_byte_data(self._address, self.REG_BLUE)
+        green = self._smbus.read_byte_data(self._address, self.REG_GREEN)
+
+        if red == blue == green == 0:
+            mode = 0
+
+        mode = self.Mode(mode)
+        status = [('Mode', str(mode), '')]
```

Comment:
I did that originally because thats how I thought that `__str__` worked, but only the number value got printed. (I just checked again and there is still this same behaviour.

Reply:
That's surprising, and not what I see when using my driver.  Needs to be debugged...